gr-qtgui/lib/eyedisplayform.cc

```diff
@@ -0,0 +1,672 @@
+            SIGNAL(signalTriggerSlope(gr::qtgui::trigger_slope)),
+            d_controlpanel,
+            SLOT(toggleTriggerSlope(gr::qtgui::trigger_slope)));
+    connect(d_stop_act, SIGNAL(triggered()), d_controlpanel, SLOT(toggleStopButton()));
+
+    d_layout->addLayout(d_controlpanel, 0, d_cols, d_rows, 1);
+
+    d_controlpanel->toggleAutoScale(d_autoscale_act->isChecked());
+    d_controlpanel->toggleGrid(d_grid_act->isChecked());
+    d_controlpanel->toggleTriggerMode(getTriggerMode());
+    d_controlpanel->toggleTriggerSlope(getTriggerSlope());
+
+    d_controlpanelmenu->setChecked(true);
+}
+
+void EyeDisplayForm::teardownControlPanel()
+{
+    if (d_controlpanel) {
+        d_layout->removeItem(d_controlpanel);
+        delete d_controlpanel;
+        d_controlpanel = NULL;
+    }
+    d_controlpanelmenu->setChecked(false);
+}
+
+EyeDisplayPlot* EyeDisplayForm::getSinglePlot(unsigned int i)
+{
+    return ((EyeDisplayPlot*)d_displays_plot[i]);
+}
+
+void EyeDisplayForm::newData(const QEvent* updateEvent)
+{
+    TimeUpdateEvent* tevent = (TimeUpdateEvent*)updateEvent;
+    const std::vector<double*> dataPoints = tevent->getTimeDomainPoints();
+    const uint64_t numDataPoints = tevent->getNumTimeDomainDataPoints();
+    const std::vector<std::vector<gr::tag_t>> tags = tevent->getTags();
+
+    for (unsigned int i = 0; i < d_nplots; ++i) {
+        getSinglePlot(i)->plotNewData(
+            dataPoints, numDataPoints, d_sps, d_update_time, tags);
+    }
+}
+
+void EyeDisplayForm::customEvent(QEvent* e)
+{
+    if (e->type() == TimeUpdateEvent::Type()) {
+        newData(e);
+    }
+}
+
+void EyeDisplayForm::setSampleRate(const QString& samprate)
+{
+    setSampleRate(samprate.toDouble());
+}
+
+void EyeDisplayForm::setSampleRate(const double samprate)
+{
+    if (samprate > 0) {
+        std::string strtime[4] = { "sec", "ms", "us", "ns" };
+        double units10 = floor(log10(samprate));
+        double units3 = std::max(floor(units10 / 3.0), 0.0);
+        double units = pow(10, (units10 - fmod(units10, 3.0)));
+        int iunit = static_cast<int>(units3);
+
+        d_current_units = units;
+
+        for (unsigned int i = 0; i < d_nplots; ++i) {
+            getSinglePlot(i)->setSampleRate(samprate, units, strtime[iunit]);
+        }
+    } else {
+        throw std::runtime_error("EyeDisplayForm: samprate must be > 0.");
+    }
+}
+
+void EyeDisplayForm::setYaxis(double min, double max)
+{
+    for (unsigned int i = 0; i < d_nplots; ++i) {
+        getSinglePlot(i)->setYaxis(min, max);
+    }
+}
+
+void EyeDisplayForm::setAxisLabels(bool en)
+{
+
+    for (unsigned int i = 0; i < d_nplots; ++i) {
+        getSinglePlot(i)->setAxisLabels(en);
+    }
+}
+
+void EyeDisplayForm::setYLabel(const std::string& label, const std::string& unit)
+{
+    for (unsigned int i = 0; i < d_nplots; ++i) {
+        getSinglePlot(i)->setYLabel(label, unit);
+    }
+}
+
+int EyeDisplayForm::getNPoints() const { return d_npoints; }
```

Comment:
Please update the formatting of these functions as detailed in a previous comment